### DISCUSSION TITLE
How to debug image-customization.lua?

### DISCUSSION THREAD

**Discussion by citronalco** [freifunk-gluon/gluon#3217] [Q&A] [answered]
We are building two firmware variants: fastd and l2tp.
In the old site.mk, we we checked environment variables to figure out whether to include fastd or tunneldigger.

Translated to Lua, this should be something similar to this:
```
os = local require 'os'
if os.getenv("VPN_TYPE") == "l2tp" then
        features({
                'mesh-vpn-tunneldigger',
        })
elseif os.getenv("VPN_TYPE") == "fastd" then
        features({
                'mesh-vpn-fastd',
        })
end
```

But this does not work. Even if I enter complete nonsense into image-customization.lua, the build process simply goes on.

How can I figure out if image-customization.lua is valid? Is there an easy way to see the the resulting values of "features" and "packages"?

**Comment by neocturne** [comment, marked as answer]
The file being ignored when it is invalid is a bug, I have created https://github.com/freifunk-gluon/gluon/issues/3218 to track this.

The first line has a syntax error (it should be `local os = require 'os'`), but it would not work anyways - `image-customization.lua` runs sandboxed, none of the standard Lua functions and modules like `require` or `os` are accessible.
